### DISCUSSION TITLE
[Optional Feature] : Perform output caching on Marked functions if the parameter is identical | reference equal

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/runtime#73111] [Ideas]
Functions can be marked by attributes or keyword, then the CLR gives the instance a captured dictionary that will be cleared unpredictably (whenever GC or whenever is convenient), the key of the dictionary would be the parameter while the value would be the function's output

**Comment by huoyaoyuan** [comment]
I don't think it's worthy to couple so much feature with the runtime.
I do considered a collection of weak reference that's cleaned up by GC. This proposal can be implemented manually then.

**Reply by Xyncgas** [reply]
If we can have static local like VB, it would be much obvious how to implement the caching manually, although regardless of how this will be done imo caching is just a way to optimize repetitive function calls if we are not doing this already, which might be a few reasons like it's hard as I would imagine for rt / compiler to confirm that repetitive function calls (with same params) always produce the same results, that's why I thought a marking strategy was necessary for this feature hence the discussion.

You've always provided insightful information so thanks for being around and helping us
